### PR TITLE
CNF-14660: ClusterRequest spec changes

### PR DESCRIPTION
**Pull request by Missxiaoguo**
- Update ClusterRequest spec fields to match
```
spec:
    name: <string, human readable name for the ProvisioningRequest> <- new
    description: <string, more detailed description provided by SMO> <- new
    templateName: <string, name of the O-Cloud template> <- has done in another PR
    templateVersion: <string,version of the O-Cloud template > <- has done in another PR
    templateParameters: <single schema> <- replaces clusterTemplateInput
        nodeClusterName:  <string> 
        oCloudSiteId:           <string>
        clusterInstanceParameters:  <parms for clusterInstance data>
        policyTemplateParameters:  <parms for policyTemplate data>
    extensions: <FFS> <- new
```
- templateParameters validation updates to ensure expected parameters

- a quick fix for clusterinstance template issue

TODO: remove timeout from spec

### CODE REVIEW

File: internal/controllers/clusterrequest_controller_test.go [hunk 1]
```diff
@@ -635,6 +643,7 @@ var _ = Describe("ClusterRequestReconcile", func() {
 
 	Context("Resources preparation during initial Provisioning", func() {
 		It("Verify status conditions if ClusterRequest validation fails", func() {
+			fmt.Println("ANGIEEe- ", testFullTemplateParameters)
```
Comment (edcdavid) [inline internal/controllers/clusterrequest_controller_test.go:646]: leftover test code

Comment (Missxiaoguo) [reply internal/controllers/clusterrequest_controller_test.go:646]: Good catch!

File: internal/controllers/utils/constants.go [hunk 2]
```diff
@@ -68,13 +68,19 @@ const (
 	defaultServiceCAFile    = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt" // nolint: gosec // hardcoded path only
 )
 
+// Required template schema parameters
+const (
+	TemplateParamNodeClusterName = "nodeClusterName"
+	TemplateParamOCloudSiteId    = "oCloudSiteId"
+	TemplateParamClusterInstance = "clusterInstanceParameters"
+	TemplateParamPolicyTemplate  = "policyTemplateParameters"
```
Comment (irinamihai) [inline internal/controllers/utils/constants.go:76]: Maybe rename `TemplateParamPolicyTemplate` to `TemplateParamPolicyConfig`? I find it a bit weird to have `template` twice in the variable name.

File: internal/controllers/provisioningrequest_controller.go [hunk 3]
```diff
@@ -386,81 +384,184 @@ func (t *provisioningRequestReconcilerTask) validateProvisioningRequestCR(ctx co
...
-		return utils.NewInputError("failed to validate ClusterTemplate input matches schema: %s", err.Error())
+		return utils.NewInputError(
+			"the provided %s does not match the schema from ClusterTemplate (%s): %w",
+			utils.TemplateParamClusterInstance, clusterTemplate.Name, err)
 	}
-	// Get the merged clusterinstance input data
-	mergedClusterInstanceData, err := t.getMergedClusterInputData(ctx, clusterTemplate, utils.ClusterInstanceDataType)
+
+	// Get the merged ClusterInstance input data
+	mergedClusterInstanceData, err := t.getMergedClusterInputData(
```
Comment (edcdavid) [inline internal/controllers/provisioningrequest_controller.go:439]: Optional: Add a comment explaining why we cannot validate the merged template at this point, something like: "The clusterInstance template defined in clusterInstanceSubSchema is only a subset of the template of the SiteConfig Clustertemplate CR. However, the content of the cluster request "clusterInstanceDefaults" includes input matching the Siteconfig Full ClusterTemplate. So at this point we cannot do further validate as some parameters might exist in the Full Siteconfig template that are not in the ORAN template. The Full Cluster instance validation is performed in the Cluster Instance CR dry run"

**Comment by Missxiaoguo** [comment]
It has been tested with latest code with the changes for CR renaming

File: internal/controllers/provisioningrequest_controller.go (hunk 3 above)
Comment (Missxiaoguo) [reply internal/controllers/provisioningrequest_controller.go:444]: Updated

File: api/provisioning/v1alpha1/provisioningrequest_types.go [hunk 4]
```diff
@@ -43,30 +47,18 @@ type ProvisioningRequestSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Template Version",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	TemplateVersion string `json:"templateVersion"`
 
-	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Cluster Template Input",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
-	ClusterTemplateInput ClusterTemplateInput `json:"clusterTemplateInput"`
+	// TemplateParameters provides the input data that conforms to the schema defined in the referenced ClusterTemplate's spec.templateParameterSchema.
```
Comment (browsell) [inline api/provisioning/v1alpha1/provisioningrequest_types.go:50]: nit, would be good to add that the template must be in openapi-v3 format

File: config/samples/v1alpha1_provisioningrequest.yaml [hunk 5]
```diff
@@ -9,18 +9,21 @@ metadata:
     app.kubernetes.io/created-by: oran-o2ims
   name: provisioningrequest-sample
 spec:
-  site: "local"
+  name: "TestEnv-SNO-Provisioning"
+  description: "Provisioning request for setting up a Single Node OpenShift (SNO) in the test environment."
   timeout:
```
Comment (browsell) [inline config/samples/v1alpha1_provisioningrequest.yaml:14]: These will need to move to the templateParameters

File: api/provisioning/v1alpha1/provisioningrequest_types.go (hunk 4 above)
Comment (Missxiaoguo) [reply api/provisioning/v1alpha1/provisioningrequest_types.go:50]: The description of templateParameterSchema field in the ClusterTemplate mentions that the schema should be openAPI v3 format, but in the provisioningRequest I can change to this
```suggestion
	// TemplateParameters provides the input data that conforms to the OpenAPI v3 schema defined in the referenced ClusterTemplate's spec.templateParameterSchema.
```

**Comment by Missxiaoguo** [comment]
/hold

Removing timeout from spec

**Comment by Missxiaoguo** [comment]
/unhold
Decide to have a followup PR for removing timeout

**Pull request merged**